src/multivariate/multinomial.jl

```diff
@@ -118,16 +122,18 @@ end
 function _logpdf{T<:Real}(d::Multinomial, x::AbstractVector{T})
     p = probs(d)
     n = ntrials(d)
-    s = lgamma(n + 1.0)
-    t = 0
+    S = eltype(p)
+    R = promote_type(T, S)
+    s = R(lgamma(n + 1))
+    t = zero(T)
     for i = 1:length(p)
         @inbounds xi = x[i]
-        @inbounds pi = p[i]
+        @inbounds p_i = p[i]
         t += xi
-        s -= lgamma(xi + 1.0)
-        @inbounds s += xi * log(pi)
+        s -= R(lgamma(xi + 1))
```

Comment:
Right. But if `xi = 1`, then 
```julia
julia> typeof(lgamma(xi + 1))
Float64
```
and if `typeof(d)` is `Multinomial{Float32}`, we need this to be `Float32`.